### PR TITLE
[BC5] Merge latest main

### PR DESCRIPTION
**Pull request by joshdholtz**
⚠️  This _still_ has merge conflicts into `main` and I have no idea what's the happening...

### Motivation

To have latest stuff from `main` for our release

### Description

Merge `main` into `bc5-support`

### CODE REVIEW

File: .run/Integration Tests.run.xml
Comment (joshdholtz) [inline .run/Integration Tests.run.xml:1]: Is this supposed to be here? 👀

File: .run/Purchases Integration Tests (set purchases module variant before executing).run.xml
Comment (joshdholtz) [inline .run/Purchases Integration Tests (set purchases module variant before executing).run.xml:1]: Is this supposed to be here? 👀

File: .run/Integration Tests.run.xml
Comment (tonidero) [reply .run/Integration Tests.run.xml:1]: These are new run configurations fro Android Studio, so it's easier to run integration tests. We have disabled them [here](https://github.com/RevenueCat/purchases-android/pull/904/files#diff-8302aedba0ad65fbbb58f3a89c2ee6520963fc8c65dba3066591b53fca8344efR8-R16) so they will not work until we fix them though

**Comment by joshdholtz** [comment]
Closing in favor of #905

**Pull request closed without merging**